lib/9pfs/9pfs_vnops.c

```diff
@@ -900,7 +905,7 @@ static int uk_9pfs_fsync(struct vnode *vp, struct vfscore_file *fp)
 
 static int uk_9pfs_truncate(struct vnode *vp, off_t off)
 {
-	return uk_9pfs_setattr(vp, &(struct vattr){
+	return uk_9pfs_do_setattr(vp, UK_9PFS_VFID(vp), &(struct vattr){
```

Comment:
Isn't this just a call to the `uk_9pfs_setattr()`, since the `fid` is `UK_9PFS_VFID(vp)`?

Reply:
Yes, you could leave it like that. Question is, if it really makes sense. Since this is a trivial direct call and `uk_9pfs_setattr()` does not do any extra work, you just increase the number of hops (=lower performance). Furthermore, this would create a dependency between `truncate` and `setattr` whereas the actual dependency is between `truncate` and `do_setattr`.

Does that make sense?

Reply:
Yes, that makes sense.
I don't have a strong opinion on this, I generally just hate having to use wrappers, but when it's needed it's needed.
All looks good then, thank you :rocket:

Reply:
Yes, it would be nice if we could avoid the `_do_` variant altogether. But since `uk_9pfs_setattr` is a vnop that does not work.